### PR TITLE
Graduate Volume Expansion to Beta + e2e tests

### LINKED ISSUE #81166: Move CSI Volume expansion feature gate to beta
We need to make sure that all the tests are enabled and make sure that we test more than just mock CSI volume plugin.

### PR DESCRIPTION
**Pull request by bertinatto**
**What type of PR is this?**

/kind feature

**What this PR does / why we need it**:

Move volume expansion support to beta and enable e2e tests.

**Which issue(s) this PR fixes**:

Fixes #81166

```release-note
Move CSI volume expansion to beta.
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs
https://github.com/kubernetes/enhancements/issues/556
```

### CODE REVIEW

**Comment by bertinatto** [comment]
This PR depends on:

https://github.com/kubernetes-csi/csi-driver-host-path/pull/87
https://github.com/kubernetes-csi/csi-driver-host-path/pull/90

File: test/e2e/storage/testsuites/volume_expand.go [hunk 1]
```diff
@@ -259,24 +259,28 @@ func (v *volumeExpandTestSuite) defineTests(driver TestDriver, pattern testpatte
...
-			return true, nil
+		pvc.Spec.Resources.Requests[v1.ResourceStorage] = size
+		updatedPVC, err := c.CoreV1().PersistentVolumeClaims(origPVC.Namespace).Update(pvc)
+		if err != nil {
+			// If an error happened, we don't return the Update()'d PVC
+			// because it might be invalid (e.g., with an empty name).
+			e2elog.Logf("Error updating pvc %s with %v", pvcName, err)
+			return false, nil
 		}
-		e2elog.Logf("Error updating pvc %s with %v", pvcName, err)
```
Comment (gnufied) [inline test/e2e/storage/testsuites/volume_expand.go]: This will be fixed once - https://github.com/kubernetes/kubernetes/pull/81107merges.

**Review by msau42 — COMMENTED** [review]
/approve

File: test/e2e/storage/testpatterns/testpattern.go [hunk 2]
```diff
@@ -219,17 +219,17 @@ var (
 
 	// DefaultFsDynamicPVAllowExpansion is TestPattern for "Dynamic PV (default fs)(allowExpansion)"
 	DefaultFsDynamicPVAllowExpansion = TestPattern{
-		Name:           "Dynamic PV (default fs)(allowExpansion)",
-		VolType:        DynamicPV,
-		BindingMode:    storagev1.VolumeBindingWaitForFirstConsumer,
+		Name:    "Dynamic PV (default fs)(allowExpansion)",
+		VolType: DynamicPV,
+		// BindingMode:    storagev1.VolumeBindingWaitForFirstConsumer,
```
[the remark this answers is not in the document]
Comment (bertinatto) [reply test/e2e/storage/testpatterns/testpattern.go]: Removed

File: test/e2e/storage/testpatterns/testpattern.go [hunk 3]
```diff
@@ -219,17 +219,17 @@ var (
...
 	// BlockVolModeDynamicPVAllowExpansion is TestPattern for "Dynamic PV (block volmode)(allowExpansion)"
 	BlockVolModeDynamicPVAllowExpansion = TestPattern{
-		Name:           "Dynamic PV (block volmode)(allowExpansion)",
-		VolType:        DynamicPV,
-		VolMode:        v1.PersistentVolumeBlock,
-		BindingMode:    storagev1.VolumeBindingWaitForFirstConsumer,
+		Name:    "Dynamic PV (block volmode)(allowExpansion)",
+		VolType: DynamicPV,
+		VolMode: v1.PersistentVolumeBlock,
+		// BindingMode:    storagev1.VolumeBindingWaitForFirstConsumer,
```
[the remark this answers is not in the document]
Comment (bertinatto) [reply test/e2e/storage/testpatterns/testpattern.go]: removed

File: test/e2e/storage/testsuites/volume_expand.go (hunk 1 above)
Comment (bertinatto) [reply test/e2e/storage/testsuites/volume_expand.go]: Dropped commit

File: pkg/features/kube_features.go [hunk 4]
```diff
@@ -108,7 +108,7 @@ const (
 	ExpandInUsePersistentVolumes featuregate.Feature = "ExpandInUsePersistentVolumes"
 
 	// owner: @gnufied
-	// alpha: v1.14
+	// beta: v1.16
```
[the remark this answers is not in the document]
Comment (bertinatto) [reply pkg/features/kube_features.go]: Added alpha line back

**Review by saad-ali — APPROVED** [review]
/lgtm
/approve

**Pull request merged**